events/bus_test.go

```diff
@@ -0,0 +1,34 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package events
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestBusPublish(t *testing.T) {
```

Comment:
suggestion: would be great to have more tests. Current one covers only the happy path and nothing else.
cases like these: subscribe twice to the same event, test unsubscribe, test if not subscribed-to event is not received, test close and others.

I'm sure it's mostly covered with integration tests, but would nice to have explicit unit tests for the event bus